.github/workflows/ci.yml

```diff
@@ -44,20 +44,21 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
-    - uses: pdm-project/setup-pdm@v4
+    - name: Setup uv
+      uses: astral-sh/setup-uv@v3
       with:
-        # cloudflare pages build uses python 3.12, see `/.python-version`
-        python-version: '3.12'
+        enable-cache: true
+
+    - name: Set up Python 3.12
+      run: uv python install 3.12
 
     - name: install dependencies
       # Unlike the docs build, we don't use mkdocs_material-insiders
       # Because the secret for accessing the library is not accessible from forks, but we still want to run
       # this job on public CI runs.
-      run: |
-        pdm venv create --with-pip --force $PYTHON
-        pdm install -G docs
+      run: uv sync --python 3.12 --group docs
```

Comment:
uv automatically checks `requires-python` constraints and use that python version. Is it necessary to mention the python version again here?

Reply:
It does seem redundant...

Reply:
So the thing is `uv` will prefer the specified version in the `.python-version` file if present, no matter the previously installed version. We currently don't have such a file, but it could be pretty bad if we end up creating one at some point, especially for the jobs with a Python version matrix: CI will only run on the version from `.python-version`, and we won't notice anything.

It's a bit unfortunate, probably using tox (and tox-uv) could help.

Reply:
when a python version matrix exists, then python version is specified with the sync command
`uv sync --python ${{ matrix.python-version }}`

So it will use the requested python version from matrix even the `.python-version` file exist